src/main/groovy/org/pih/warehouse/monitoring/GrailsHttpTransactionNameProvider.groovy

```diff
@@ -0,0 +1,23 @@
+package org.pih.warehouse.monitoring
+
+import io.sentry.protocol.TransactionNameSource
+import io.sentry.spring.tracing.TransactionNameProvider
+import javax.servlet.http.HttpServletRequest
+
+/**
+ * Defines how Sentry transaction names for HTTP requests should be structured.
+ */
+class GrailsHttpTransactionNameProvider implements TransactionNameProvider {
+
+    String provideTransactionName(HttpServletRequest var1) {
+        // If we wanted our logs to look more like HTTP Requests, we could have done the following instead, but
+        // we decided that "invoiceApi/list" is easier to debug than "GET /openboxes/api/invoices".
+        //return "${var1.method} ${var1.requestURI}"
+        return "${var1.getAttribute("org.grails.CONTROLLER_NAME_ATTRIBUTE")}/${var1.getAttribute("org.grails.ACTION_NAME_ATTRIBUTE")}"
```

Comment:
I understand that invoiceApi/list is easier to parse and figure out where to look, but using the URL Mapping seems to be a better choice. 

Is there a way to make the choice configurable? I assume the easiest way would be to leave this class alone and handle the configuration in the SentryGrailsTracingFilter.

Note: I am not asking you to do it, just wondering what it would entail.

Reply:
I like this approach better (implementing a Sentry transaction name provider) but just wanted to point out that we're setting the transaction name for New Relic here.

https://github.com/openboxes/openboxes/blob/d0a8cab0ba9a4333529c05e39aac3515ddaffe39/grails-app/conf/openboxes/NewRelicFilters.groovy

Perhaps someday we'll want the transaction names to be consistent.

Reply:
Yeah we could make it configurable via an app setting. I already have the code for both options so I have no problem adding a property for this.